### PR TITLE
Describe architecture of Director

### LINKED ISSUE #57: Design Registry API
**AC**
- API layer for storage access
- update architecture document

### PR DESCRIPTION
**Pull request by pkosiec**
**Description**

Changes proposed in this pull request:

- Describe architecture of Director

**Related issue(s)**
Resolves #57

### CODE REVIEW

File: docs/architecture/director-component.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+![](./assets/director-architecture.svg)
+
+### Root Resolver
+
+Root Resolver is a type, which composes of Domain Resolver types. In a result, Root Resolver consists of all GraphQL resolvers needed to run the GraphQL server.  
+
+### Domain Resolver
+
+Domain Resolver is a set of GraphQL resolvers related to a specific domain. For example, Runtime Domain Resolver contains all GraphQL resolvers related to Runtime type, including queries, mutations and field resolvers.
```
Comment (crabtree) [inline docs/architecture/director-component.md]: I would rephrase this. Currently, on the diagram, there is no Domain Resolver or Runtime Domain Resolver. I would say that Runtime Resolver is one of Domain Resolver types.

File: docs/architecture/director-component.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# Director Component
+
+## Overview
+
+This document describes [Director](../terminology.md)'s architecture.
+
+## Architecture
+
+The following diagram illustrates the overall architecture of this application:
```
Comment (crabtree) [inline docs/architecture/director-component.md]: I have some doubt whether the services and repositories should be drawn as a part of Root Resolver. In general, those components should not be aware that the layer which calls them is a GraphQL or anything else.

Comment (pkosiec) [reply docs/architecture/director-component.md]: Hmm, technically it's composition and that's what I wanted to show:
Root Resolver consists of Domain Resolvers, Domain Resolvers consists of Services, and Services of Repositories. 🤔

File: docs/architecture/director-component.md (hunk 1 above)
Comment (pkosiec) [reply docs/architecture/director-component.md]: To be honest firstly I wanted to create "generic" diagram, with Domain 1 Resolver, Domain 1 Service etc. What do you think - would it be better?

Comment (crabtree) [reply docs/architecture/director-component.md]: That may be a better approach as our components are not a fixed set.

File: docs/architecture/director-component.md (hunk 2 above)
Comment (crabtree) [reply docs/architecture/director-component.md]: Maybe just do not enclose it in one block

Comment (pkosiec) [reply docs/architecture/director-component.md]: @crabtree What about now?

File: docs/architecture/director-component.md (hunk 1 above)
Comment (pkosiec) [reply docs/architecture/director-component.md]: @crabtree What about now?

File: docs/architecture/director-component.md (hunk 2 above)
Comment (crabtree) [reply docs/architecture/director-component.md]: 👍

File: docs/architecture/director-component.md (hunk 1 above)
Comment (crabtree) [reply docs/architecture/director-component.md]: 👍

File: docs/architecture/director-component.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+Domain Resolver uses Services. To convert from domain model to GraphQL types, Domain Resolver uses converters, which are types that contain conversion methods.
+
+### Service
+
+Service contains business logic related to a specific domain. Single Resolver can use multiple Services from different domain packages. To do CRUD operations, Service uses Repository.
+
+### Repository
+
+Repository is an abstraction over storage access. It is used for CRUD operations on database. In future, it may be separated from the Director as an additional component.
```
Comment (PK85) [inline docs/architecture/director-component.md]: Maybe already mention about Registry: "..  as an additional component called Registry."
